### ISSUE TITLE
Not always picking up body background in color-contrast test

### ISSUE THREAD

**Issue by badtant** [dequelabs/axe-core#560]
Hi! First of all, thanks for a great tool.

I get two color-contrast violations on this page (with the chrome extension):
https://www.svtplay.se/klipp/15174068/bill-skarsgard-spelar-skrackclown-i-nya-filmen-det

The messages say that the background is #ffffff though #121314 is set on the body element. So for some reason the body element is not picked up as a parent and checked for background. Clearly this must be a bug.

Thanks

**Comment by dylanb** [comment]
How are you running the rules because when I run the aXe extension against that page, it doesn't flag anything

**Comment by badtant** [comment]
That sounds really odd. I'm just clicking "analyze" in the chrome extension. 
The Firefox addon doen't report anything though, but that's some other problem :)

**Comment by marcysutton** [comment]
I can confirm that is a bug. In the latest aXe extension, I get two color violations with a background color of #ffffff, which is the HTML background. The body element has a background-color of #121314, which isn't being picked up for some reason due to the layout. If I turn off the `height: 100%` in Chrome devtools and run again, the violations are no longer reported. It's potentially a bug related to `document.elementsFromPoint`. If it isn't reporting this one in Firefox, that's because the bug is only reproducible in Chrome.

It's also worth pointing out that if it was a "review item" rather than a violation, it wouldn't show in Firefox because that extension UI hasn't yet been updated to show review items. But this one is related to a color contrast bug in Chrome.

**Comment by dylanb** [comment]
![image](https://user-images.githubusercontent.com/79245/31291583-23dbacf4-aa85-11e7-81fb-0b02aebdddd1.png)

**Comment by badtant** [comment]
@marcysutton thanks for confirming
@dylanb really really strange. I get 2 violations and 40 review

**Comment by marcysutton** [comment]
Super odd. I am running the same extension version, in Chrome 61.0.3163.100. I get 40 review items in the default state of the page, and the 2 color violations. If I uncheck `height: 100%` on `html, body`, I get 42 review items with 40 of them color contrast and 2 for HTML5 video.

**Comment by badtant** [comment]
I get exactly the same as Marcy (with dev tools docked to bottom)

**Comment by marcysutton** [comment]
@badtant I no longer get a violation for the body element on this page–there are 14 review items. Can you tell if the page updated, and if so, provide us with an isolated case to test against? Otherwise I'm going to close this issue, as we can't reproduce it.

**Comment by badtant** [comment]
@marcysutton Seems like they just changed something. I still get the false errors on the startpage https://www.svtplay.se/ Both in Chrome and FF.

**Comment by DiegoBM-DB** [comment]
I presume this error never got solved since it's happening to me the same, it skips the body color and reports the html color as the background-color for text elements

**Comment by marcysutton** [comment]
@DiegoBM-DB is your problem with this same page? If not, can you please open a new issue? There are many reasons why an element's background color can fall through the stack, and your case might not be the same as this one. 

In looking at https://www.svtplay.se/, the color contrast violation does appear to be wrong for nodes 1-18, and there are no related nodes (but there should be at least one). It isn't picking up the body background color of `#0b0c0d` for some reason:
```
<p class="play_content-item__subtext">Säsong 35 - Farnborough 2:2</p>
```
There isn't any obvious reason to me why that element's background doesn't stick around while the h2 next to it is fine; it will require stepping through the entire color algorithm to see where the body gets filtered out.

**Comment by DiegoBM-DB** [comment]
Hi Marcy, thank you for the quick response. It's a different web page. I thought I could reuse an existing item in order to not create duplicates which are widely hated in Github, since it seems to have the very same symptoms:  background is reported to be #ffffff even though background-color #0f131 is assigned in the body tag. If I assign that same color in the html tag, the error goes away. But if you think is the right thing to do, I'll create a new issue

**Comment by marcysutton** [comment]
@DiegoBM-DB does your body element have `height: 100%`?

**Comment by DiegoBM-DB** [comment]
@marcysutton it does not, but I reckon I have found the culprit. The element right below <body> is a div which contains the entire web application and it's positioned absolute. If I remove that positioning the error goes away

html
  body
    div-abolute-positioned
       ....
       restof the web app
       ....

**Comment by WilcoFiers** [comment]
I was able to reproduce this issue with axe-core 2.1.x and 2.2.x, it seems to have been fixed as of axe-core 2.3.x.
